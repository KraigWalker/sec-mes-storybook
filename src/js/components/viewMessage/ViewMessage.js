import React from 'react';
import StepHeader from '../common/StepHeader';
import SecureMessageSummary from '../common/SecureMessageSummary';
import TextArea from '../common/TextAreaComponent';
import Threads from '../common/ThreadList'
import { connect } from 'react-redux';
import { getSecureMessages } from '../../actions/AppActions';
import {getThreadsBL} from '../../bl/SecureMessageBL'
import {getMessageType} from '../../utils/SecureMessageUtils';

class ViewMessage extends React.Component {
    componentWillMount(){
        this.props.dispatch(getSecureMessages());
    }
    
    getThreads(messages, currentMessage){
        const threads =  getThreadsBL(messages, currentMessage);
        return <Threads Threads={threads} />
    }

    render() {
        const { messageDetail} = this.props.location;
        return (
            <div className="container">
                <div className="row">
                    <div className="col-md1-18">
                        <StepHeader showheaderCrumbs={true} headerCrumbsPath={{ pathname : '/securemessages'}} headerCrumbsMessage="Back" 
                        headerTitle={(getMessageType(messageDetail.status)== 'sent')?this.props.headerDetails.senttitle:this.props.headerDetails.inboxtitle}/>
                    </div>
                </div>
                    <SecureMessageSummary message= { messageDetail }></SecureMessageSummary>
                    <div className="c-field">
                        <TextArea messageBody={messageDetail.messageBody} disableText={true}/>
                    </div>
                    {this.getThreads(this.props.messages.messages, messageDetail)}
                
            </div>
        );
    }
}


const mapState = (state) => { 
    return {
        messages: state.messages
    }
};


export default connect(mapState)(ViewMessage);
