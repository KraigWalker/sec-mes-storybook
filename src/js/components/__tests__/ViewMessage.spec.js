import React from 'react';
import { ViewMessage } from '../ViewMessage';
import { shallow } from 'enzyme';
import TestUtils from 'react-addons-test-utils';
jest.mock('../../bl/SecureMessageBL');

describe("View Message snapshot", () => {
    const dispatch = jest.fn();
    const filter = jest.fn();
    let props = {
        content: {
            back: 'Back',
        },
        activeTab: 'SENT',
        messages: {
            draftError: true,
            successModal: true,
            filter: filter,
        },
        location: {
            messageDetail: {
                status: 'NEW',
            }
        },
        messagesFetched: {
            fetching: false,
            successModal: false
        },
        dispatch: dispatch
    };
    let component = shallow(<ViewMessage {...props} />);
    it('should match to snapshot', () => {
        component.setState({ showDeleteConfirmModal: true, showDeleteSuccessModal: true, showSendServiceErrorModal: true });
        expect(component).toMatchSnapshot();
    });
    it('CloseModal function test', () => {
        component.instance().closeModal();
        expect(component.instance().state.showDeleteConfirmModal).toBeFalsy();
    });
    it('errorCloseClicked function test', () => {
        component.instance().errorCloseClicked();
        expect(component.instance().state.showSendServiceErrorModal).toBeFalsy();
    });
    it('handleDelete function test', () => {
        component.instance().handleDelete();
        expect(component.instance().state.showDeleteConfirmModal).toBeTruthy();
    });
    it('deleteClick function test', () => {
        component.instance().deleteClick();
        expect(component.instance().state.showDeleteConfirmModal).toBeFalsy();
    });
    it('closeSuccessModal function test', () => {
        component.instance().closeSuccessModal();
        expect(component.instance().state.showDeleteSuccessModal).toBeFalsy();
    });
    it('retryServiceCall function test', () => {
        component.instance().retryServiceCall();
        expect(props.dispatch).toBeCalled();
    });
    it('renders attachment section if hasAttachment true', () => {
        let props = {
            content: {
                sentPageTitle: 'sentPageTitle',
            },
            activeTab: 'SENT',
            messages: {
                draftError: true,
                successModal: true,
                filter: filter,
            },
            location: {
                messageDetail: { status: 'sent' }
            },
            messagesFetched: {
                fetching: false,
                successModal: false
            },
            dispatch: dispatch,
            hasAttachment: true
        };
        const component = shallow(<ViewMessage {...props} />);
        expect(component.find("Attachments")).toHaveLength(1);
    });

    it("does not render attachments if hasAttachment false", () => {
        let props = {
            content: {
                sentPageTitle: 'sentPageTitle',
            },
            activeTab: 'SENT',
            messages: {
                draftError: true,
                successModal: true,
                filter: filter,
            },
            location: {
                messageDetail: { status: 'sent' }
            },
            messagesFetched: {
                fetching: false,
                successModal: false
            },
            dispatch: dispatch,
            hasAttachment: false
        };
        const component = shallow(<ViewMessage {...props} />);
        expect(component.find("Attachments")).toHaveLength(0);
    });

    it("updates message status to READ if NEW message", () => {
        let props = {
            content: {
                sentPageTitle: 'sentPageTitle',
            },
            activeTab: 'SENT',
            messages: {
                draftError: true,
                successModal: true,
                filter: filter,
            },
            location: {
                messageDetail: { status: 'NEW' }
            },
            messagesFetched: {
                fetching: false,
                successModal: false
            },
            dispatch: dispatch,
            setMessagesMetaData: jest.fn(),
            hasAttachment: false,
            isWebView: false,
        };
        shallow(<ViewMessage {...props} />);
        expect(dispatch).toHaveBeenCalled();
    });

    it("updates native app with new unread message count if within webView", () => {
        let props = {
            content: {
                sentPageTitle: 'sentPageTitle',
            },
            activeTab: 'SENT',
            messages: {
                draftError: true,
                successModal: true,
                filter: filter,
                messages: [
                    { status: 'NEW' }, { status: 'NEW' }, { status: 'NEW' }
                ]
            },
            location: {
                messageDetail: { status: 'NEW' }
            },
            messagesFetched: {
                fetching: false,
                successModal: false
            },
            dispatch: dispatch,
            setMessagesMetaData: jest.fn(),
            hasAttachment: false,
            isWebView: true,
        };
        shallow(<ViewMessage {...props} />);
        expect(props.setMessagesMetaData).toHaveBeenCalledWith({
            unread: 2
        });
    });

    it("does not update native app with new unread messafge count if not within web view", () => {
        let props = {
            content: {
                sentPageTitle: 'sentPageTitle',
            },
            activeTab: 'SENT',
            messages: {
                draftError: true,
                successModal: true,
                filter: filter,
                messages: [
                    { status: 'NEW' }, { status: 'NEW' }, { status: 'NEW' }
                ]
            },
            location: {
                messageDetail: { status: 'NEW' }
            },
            messagesFetched: {
                fetching: false,
                successModal: false
            },
            dispatch: dispatch,
            setMessagesMetaData: jest.fn(),
            hasAttachment: false,
            isWebView: false,
        };
        shallow(<ViewMessage {...props} />);
        expect(props.setMessagesMetaData).not.toHaveBeenCalled();
    });

    it("does not update message status to read when in readOnly mode", () => {
        let props = {
            content: {
                sentPageTitle: 'sentPageTitle',
            },
            activeTab: 'SENT',
            messages: {
                draftError: true,
                successModal: true,
                filter: filter,
                messages: [
                    { status: 'NEW' }, { status: 'NEW' }, { status: 'NEW' }
                ]
            },
            location: {
                messageDetail: { status: 'NEW' }
            },
            messagesFetched: {
                fetching: false,
                successModal: false
            },
            dispatch: jest.fn(),
            setMessagesMetaData: jest.fn(),
            hasAttachment: false,
            isWebView: false,
            readOnly: true,
        };
        shallow(<ViewMessage {...props} />);
        expect(props.dispatch).toHaveBeenCalledTimes(2);
    });

    it("does not render Reply button if message has noReply flag", () => {
        let props = {
            content: {
                sentPageTitle: 'sentPageTitle',
            },
            activeTab: 'SENT',
            messages: {
                draftError: true,
                successModal: true,
                filter: filter,
                messages: [
                    { status: 'NEW' }, { status: 'NEW' }, { status: 'NEW' }
                ]
            },
            location: {
                messageDetail: { status: 'NEW', noReply: true }
            },
            messagesFetched: {
                fetching: false,
                successModal: false
            },
            dispatch: jest.fn(),
            setMessagesMetaData: jest.fn(),
            hasAttachment: false,
            isWebView: false,
            readOnly: false,
        };
        const wrapper = shallow(<ViewMessage {...props} />);
        expect(wrapper.find("#reply-button")).toHaveLength(0);
    });

    it("renders reply button if not readOnly mode and no noReply flag", () => {
        let props = {
            content: {
                sentPageTitle: 'sentPageTitle',
            },
            activeTab: 'SENT',
            messages: {
                draftError: true,
                successModal: true,
                filter: filter,
                messages: [
                    { status: 'NEW' }, { status: 'NEW' }, { status: 'NEW' }
                ]
            },
            location: {
                messageDetail: { status: 'NEW' , noReply: false}
            },
            messagesFetched: {
                fetching: false,
                successModal: false
            },
            dispatch: jest.fn(),
            setMessagesMetaData: jest.fn(),
            hasAttachment: false,
            isWebView: false,
            readOnly: false,
        };
        const wrapper = shallow(<ViewMessage {...props} />);
        expect(wrapper.find("#reply-button")).toHaveLength(1);
    });

});
describe("Main snapshot", () => {
    const dispatch = jest.fn();
    const filter = jest.fn();
    let props = {
        content: {
            sentPageTitle: 'sentPageTitle',
        },
        activeTab: 'SENT',
        messages: {
            draftError: true,
            successModal: true,
            filter: filter,
        },
        location: {
            messageDetail: { status: 'sent' }
        },
        messagesFetched: {
            fetching: false,
            successModal: false
        },
        dispatch: dispatch
    };
    let component = shallow(<ViewMessage {...props} />);
    it('should match to snapshot', () => {
        component.setState({ showDeleteConfirmModal: false, showDeleteSuccessModal: false, showSendServiceErrorModal: false });
        expect(component).toMatchSnapshot();
    });
});