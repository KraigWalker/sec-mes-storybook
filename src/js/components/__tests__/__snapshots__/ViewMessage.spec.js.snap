// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewMessage
    activeTab="SENT"
    content={
      Object {
        "sentPageTitle": "sentPageTitle",
      }
    }
    location={
      Object {
        "messageDetail": Object {
          "status": "sent",
        },
      }
    }
    messages={
      Object {
        "draftError": true,
        "filter": [MockFunction],
        "successModal": true,
      }
    }
    messagesFetched={
      Object {
        "fetching": false,
        "successModal": false,
      }
    }
    setViewMessageDetail={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "status": "sent",
            },
          ],
        ],
      }
    }
    successModal={true}
    updateMessageData={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Row>
        <Card
          bordered={false}
        >
          <SectionHeading
            className="u-margin-bottom-2"
            heading1={undefined}
          />
          <Connect(withRouter(withMessaging))
            activeTab="SENT"
            content={
              Object {
                "sentPageTitle": "sentPageTitle",
              }
            }
            hasAttachment={undefined}
            location={
              Object {
                "messageDetail": Object {
                  "status": "sent",
                },
              }
            }
            messages={
              Object {
                "draftError": true,
                "filter": [MockFunction],
                "successModal": true,
              }
            }
            messagesFetched={
              Object {
                "fetching": false,
                "successModal": false,
              }
            }
            newMessageStatus="sent"
            setViewMessageDetail={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "status": "sent",
                    },
                  ],
                ],
              }
            }
            successModal={true}
            updateMessageData={[MockFunction]}
          />
        </Card>
      </Row>,
      "size": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Card
          bordered={false}
        >
          <SectionHeading
            className="u-margin-bottom-2"
            heading1={undefined}
          />
          <Connect(withRouter(withMessaging))
            activeTab="SENT"
            content={
              Object {
                "sentPageTitle": "sentPageTitle",
              }
            }
            hasAttachment={undefined}
            location={
              Object {
                "messageDetail": Object {
                  "status": "sent",
                },
              }
            }
            messages={
              Object {
                "draftError": true,
                "filter": [MockFunction],
                "successModal": true,
              }
            }
            messagesFetched={
              Object {
                "fetching": false,
                "successModal": false,
              }
            }
            newMessageStatus="sent"
            setViewMessageDetail={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "status": "sent",
                    },
                  ],
                ],
              }
            }
            successModal={true}
            updateMessageData={[MockFunction]}
          />
        </Card>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bordered": false,
          "children": Array [
            <SectionHeading
              className="u-margin-bottom-2"
              heading1={undefined}
            />,
            <Connect(withRouter(withMessaging))
              activeTab="SENT"
              content={
                Object {
                  "sentPageTitle": "sentPageTitle",
                }
              }
              hasAttachment={undefined}
              location={
                Object {
                  "messageDetail": Object {
                    "status": "sent",
                  },
                }
              }
              messages={
                Object {
                  "draftError": true,
                  "filter": [MockFunction],
                  "successModal": true,
                }
              }
              messagesFetched={
                Object {
                  "fetching": false,
                  "successModal": false,
                }
              }
              newMessageStatus="sent"
              setViewMessageDetail={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "sent",
                      },
                    ],
                  ],
                }
              }
              successModal={true}
              updateMessageData={[MockFunction]}
            />,
            Array [],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "u-margin-bottom-2",
              "heading1": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeTab": "SENT",
              "content": Object {
                "sentPageTitle": "sentPageTitle",
              },
              "hasAttachment": undefined,
              "location": Object {
                "messageDetail": Object {
                  "status": "sent",
                },
              },
              "messages": Object {
                "draftError": true,
                "filter": [MockFunction],
                "successModal": true,
              },
              "messagesFetched": Object {
                "fetching": false,
                "successModal": false,
              },
              "newMessageStatus": "sent",
              "setViewMessageDetail": [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "status": "sent",
                    },
                  ],
                ],
              },
              "successModal": true,
              "updateMessageData": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Row>
          <Card
            bordered={false}
          >
            <SectionHeading
              className="u-margin-bottom-2"
              heading1={undefined}
            />
            <Connect(withRouter(withMessaging))
              activeTab="SENT"
              content={
                Object {
                  "sentPageTitle": "sentPageTitle",
                }
              }
              hasAttachment={undefined}
              location={
                Object {
                  "messageDetail": Object {
                    "status": "sent",
                  },
                }
              }
              messages={
                Object {
                  "draftError": true,
                  "filter": [MockFunction],
                  "successModal": true,
                }
              }
              messagesFetched={
                Object {
                  "fetching": false,
                  "successModal": false,
                }
              }
              newMessageStatus="sent"
              setViewMessageDetail={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "sent",
                      },
                    ],
                  ],
                }
              }
              successModal={true}
              updateMessageData={[MockFunction]}
            />
          </Card>
        </Row>,
        "size": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Card
            bordered={false}
          >
            <SectionHeading
              className="u-margin-bottom-2"
              heading1={undefined}
            />
            <Connect(withRouter(withMessaging))
              activeTab="SENT"
              content={
                Object {
                  "sentPageTitle": "sentPageTitle",
                }
              }
              hasAttachment={undefined}
              location={
                Object {
                  "messageDetail": Object {
                    "status": "sent",
                  },
                }
              }
              messages={
                Object {
                  "draftError": true,
                  "filter": [MockFunction],
                  "successModal": true,
                }
              }
              messagesFetched={
                Object {
                  "fetching": false,
                  "successModal": false,
                }
              }
              newMessageStatus="sent"
              setViewMessageDetail={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "sent",
                      },
                    ],
                  ],
                }
              }
              successModal={true}
              updateMessageData={[MockFunction]}
            />
          </Card>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bordered": false,
            "children": Array [
              <SectionHeading
                className="u-margin-bottom-2"
                heading1={undefined}
              />,
              <Connect(withRouter(withMessaging))
                activeTab="SENT"
                content={
                  Object {
                    "sentPageTitle": "sentPageTitle",
                  }
                }
                hasAttachment={undefined}
                location={
                  Object {
                    "messageDetail": Object {
                      "status": "sent",
                    },
                  }
                }
                messages={
                  Object {
                    "draftError": true,
                    "filter": [MockFunction],
                    "successModal": true,
                  }
                }
                messagesFetched={
                  Object {
                    "fetching": false,
                    "successModal": false,
                  }
                }
                newMessageStatus="sent"
                setViewMessageDetail={
                  [MockFunction] {
                    "calls": Array [
                      Array [
                        Object {
                          "status": "sent",
                        },
                      ],
                    ],
                  }
                }
                successModal={true}
                updateMessageData={[MockFunction]}
              />,
              Array [],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "u-margin-bottom-2",
                "heading1": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeTab": "SENT",
                "content": Object {
                  "sentPageTitle": "sentPageTitle",
                },
                "hasAttachment": undefined,
                "location": Object {
                  "messageDetail": Object {
                    "status": "sent",
                  },
                },
                "messages": Object {
                  "draftError": true,
                  "filter": [MockFunction],
                  "successModal": true,
                },
                "messagesFetched": Object {
                  "fetching": false,
                  "successModal": false,
                },
                "newMessageStatus": "sent",
                "setViewMessageDetail": [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "sent",
                      },
                    ],
                  ],
                },
                "successModal": true,
                "updateMessageData": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`View Message snapshot should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewMessage
    activeTab="SENT"
    content={
      Object {
        "back": "Back",
      }
    }
    dispatch={[MockFunction]}
    location={
      Object {
        "messageDetail": Object {
          "status": "NEW",
        },
      }
    }
    messages={
      Object {
        "draftError": true,
        "filter": [MockFunction],
        "successModal": true,
      }
    }
    messagesFetched={
      Object {
        "fetching": false,
        "successModal": false,
      }
    }
    setViewMessageDetail={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "status": "NEW",
            },
          ],
        ],
      }
    }
    updateMessageData={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "status": "NEW",
            },
            undefined,
            "READ",
          ],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Row>
        <Card
          bordered={false}
        >
          <SectionHeading
            className="u-margin-bottom-2"
            heading1={undefined}
          />
          <Connect(withRouter(withMessaging))
            activeTab="SENT"
            content={
              Object {
                "back": "Back",
              }
            }
            dispatch={[MockFunction]}
            hasAttachment={undefined}
            location={
              Object {
                "messageDetail": Object {
                  "status": "NEW",
                },
              }
            }
            messages={
              Object {
                "draftError": true,
                "filter": [MockFunction],
                "successModal": true,
              }
            }
            messagesFetched={
              Object {
                "fetching": false,
                "successModal": false,
              }
            }
            newMessageStatus="READ"
            setViewMessageDetail={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "status": "NEW",
                    },
                  ],
                ],
              }
            }
            updateMessageData={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "status": "NEW",
                    },
                    undefined,
                    "READ",
                  ],
                ],
              }
            }
          />
        </Card>
      </Row>,
      "size": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Card
          bordered={false}
        >
          <SectionHeading
            className="u-margin-bottom-2"
            heading1={undefined}
          />
          <Connect(withRouter(withMessaging))
            activeTab="SENT"
            content={
              Object {
                "back": "Back",
              }
            }
            dispatch={[MockFunction]}
            hasAttachment={undefined}
            location={
              Object {
                "messageDetail": Object {
                  "status": "NEW",
                },
              }
            }
            messages={
              Object {
                "draftError": true,
                "filter": [MockFunction],
                "successModal": true,
              }
            }
            messagesFetched={
              Object {
                "fetching": false,
                "successModal": false,
              }
            }
            newMessageStatus="READ"
            setViewMessageDetail={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "status": "NEW",
                    },
                  ],
                ],
              }
            }
            updateMessageData={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "status": "NEW",
                    },
                    undefined,
                    "READ",
                  ],
                ],
              }
            }
          />
        </Card>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bordered": false,
          "children": Array [
            <SectionHeading
              className="u-margin-bottom-2"
              heading1={undefined}
            />,
            <Connect(withRouter(withMessaging))
              activeTab="SENT"
              content={
                Object {
                  "back": "Back",
                }
              }
              dispatch={[MockFunction]}
              hasAttachment={undefined}
              location={
                Object {
                  "messageDetail": Object {
                    "status": "NEW",
                  },
                }
              }
              messages={
                Object {
                  "draftError": true,
                  "filter": [MockFunction],
                  "successModal": true,
                }
              }
              messagesFetched={
                Object {
                  "fetching": false,
                  "successModal": false,
                }
              }
              newMessageStatus="READ"
              setViewMessageDetail={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "NEW",
                      },
                    ],
                  ],
                }
              }
              updateMessageData={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "NEW",
                      },
                      undefined,
                      "READ",
                    ],
                  ],
                }
              }
            />,
            Array [],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "u-margin-bottom-2",
              "heading1": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeTab": "SENT",
              "content": Object {
                "back": "Back",
              },
              "dispatch": [MockFunction],
              "hasAttachment": undefined,
              "location": Object {
                "messageDetail": Object {
                  "status": "NEW",
                },
              },
              "messages": Object {
                "draftError": true,
                "filter": [MockFunction],
                "successModal": true,
              },
              "messagesFetched": Object {
                "fetching": false,
                "successModal": false,
              },
              "newMessageStatus": "READ",
              "setViewMessageDetail": [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "status": "NEW",
                    },
                  ],
                ],
              },
              "updateMessageData": [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "status": "NEW",
                    },
                    undefined,
                    "READ",
                  ],
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Row>
          <Card
            bordered={false}
          >
            <SectionHeading
              className="u-margin-bottom-2"
              heading1={undefined}
            />
            <Connect(withRouter(withMessaging))
              activeTab="SENT"
              content={
                Object {
                  "back": "Back",
                }
              }
              dispatch={[MockFunction]}
              hasAttachment={undefined}
              location={
                Object {
                  "messageDetail": Object {
                    "status": "NEW",
                  },
                }
              }
              messages={
                Object {
                  "draftError": true,
                  "filter": [MockFunction],
                  "successModal": true,
                }
              }
              messagesFetched={
                Object {
                  "fetching": false,
                  "successModal": false,
                }
              }
              newMessageStatus="READ"
              setViewMessageDetail={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "NEW",
                      },
                    ],
                  ],
                }
              }
              updateMessageData={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "NEW",
                      },
                      undefined,
                      "READ",
                    ],
                  ],
                }
              }
            />
          </Card>
        </Row>,
        "size": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Card
            bordered={false}
          >
            <SectionHeading
              className="u-margin-bottom-2"
              heading1={undefined}
            />
            <Connect(withRouter(withMessaging))
              activeTab="SENT"
              content={
                Object {
                  "back": "Back",
                }
              }
              dispatch={[MockFunction]}
              hasAttachment={undefined}
              location={
                Object {
                  "messageDetail": Object {
                    "status": "NEW",
                  },
                }
              }
              messages={
                Object {
                  "draftError": true,
                  "filter": [MockFunction],
                  "successModal": true,
                }
              }
              messagesFetched={
                Object {
                  "fetching": false,
                  "successModal": false,
                }
              }
              newMessageStatus="READ"
              setViewMessageDetail={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "NEW",
                      },
                    ],
                  ],
                }
              }
              updateMessageData={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "NEW",
                      },
                      undefined,
                      "READ",
                    ],
                  ],
                }
              }
            />
          </Card>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bordered": false,
            "children": Array [
              <SectionHeading
                className="u-margin-bottom-2"
                heading1={undefined}
              />,
              <Connect(withRouter(withMessaging))
                activeTab="SENT"
                content={
                  Object {
                    "back": "Back",
                  }
                }
                dispatch={[MockFunction]}
                hasAttachment={undefined}
                location={
                  Object {
                    "messageDetail": Object {
                      "status": "NEW",
                    },
                  }
                }
                messages={
                  Object {
                    "draftError": true,
                    "filter": [MockFunction],
                    "successModal": true,
                  }
                }
                messagesFetched={
                  Object {
                    "fetching": false,
                    "successModal": false,
                  }
                }
                newMessageStatus="READ"
                setViewMessageDetail={
                  [MockFunction] {
                    "calls": Array [
                      Array [
                        Object {
                          "status": "NEW",
                        },
                      ],
                    ],
                  }
                }
                updateMessageData={
                  [MockFunction] {
                    "calls": Array [
                      Array [
                        Object {
                          "status": "NEW",
                        },
                        undefined,
                        "READ",
                      ],
                    ],
                  }
                }
              />,
              Array [],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "u-margin-bottom-2",
                "heading1": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeTab": "SENT",
                "content": Object {
                  "back": "Back",
                },
                "dispatch": [MockFunction],
                "hasAttachment": undefined,
                "location": Object {
                  "messageDetail": Object {
                    "status": "NEW",
                  },
                },
                "messages": Object {
                  "draftError": true,
                  "filter": [MockFunction],
                  "successModal": true,
                },
                "messagesFetched": Object {
                  "fetching": false,
                  "successModal": false,
                },
                "newMessageStatus": "READ",
                "setViewMessageDetail": [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "NEW",
                      },
                    ],
                  ],
                },
                "updateMessageData": [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "status": "NEW",
                      },
                      undefined,
                      "READ",
                    ],
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
